DependencyInjection/Compiler/ProviderCompilerPass.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Swarrot\SwarrotBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class ProviderCompilerPass implements CompilerPassInterface
+{
+    /** {@inheritDoc} */
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->has('swarrot.factory.default') || !$container->hasParameter('swarrot.provider_config')) {
+            return;
+        }
+
+        $providersIds = [];
+
+        foreach ($container->findTaggedServiceIds('swarrot.provider_factory') as $id => $tags) {
+            foreach ($tags as $tag) {
+                $providersIds[isset($tag['alias']) ? $tag['alias'] : $id] = $id;
```

Comment:
I didn't see the `id` as a fallback for the `alias`. But the `alias` as a shortcut of the `id`.
As in Symfony's FormPass: https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php#L40

I think, it's simpler for the end user to choose beetween declaring the whole serviceId or an alias.
What do you think @odolbeau ?